src/main/kotlin/org/rust/ide/console/RsConsoleRunner.kt

```diff
@@ -0,0 +1,227 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.console
+
+import com.intellij.execution.ExecutionException
+import com.intellij.execution.ExecutionManager
+import com.intellij.execution.Executor
+import com.intellij.execution.configurations.GeneralCommandLine
+import com.intellij.execution.configurations.PtyCommandLine
+import com.intellij.execution.process.OSProcessHandler
+import com.intellij.execution.runners.AbstractConsoleRunnerWithHistory
+import com.intellij.execution.ui.RunContentDescriptor
+import com.intellij.ide.errorTreeView.NewErrorTreeViewPanel
+import com.intellij.openapi.actionSystem.ActionManager
+import com.intellij.openapi.actionSystem.AnAction
+import com.intellij.openapi.actionSystem.DefaultActionGroup
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.application.ModalityState
+import com.intellij.openapi.application.TransactionGuard
+import com.intellij.openapi.diagnostic.Logger
+import com.intellij.openapi.fileEditor.FileDocumentManager
+import com.intellij.openapi.progress.ProgressIndicator
+import com.intellij.openapi.progress.ProgressManager
+import com.intellij.openapi.progress.Task
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.util.text.StringUtil
+import com.intellij.ui.GuiUtils
+import com.intellij.util.ui.MessageCategory
+import com.intellij.util.ui.UIUtil
+import org.rust.cargo.project.model.cargoProjects
+import org.rust.cargo.runconfig.command.workingDirectory
+import org.rust.openapiext.withWorkDirectory
+import java.awt.BorderLayout
+import java.nio.charset.StandardCharsets
+import java.nio.file.Paths
+import javax.swing.JPanel
+
+interface RsConsoleRunner {
```

Comment:
Do you think there will be other implementations in addition to `RsConsoleRunnerImpl`? If no, let's get rid of the interface.

Reply:
I think there could be action "run selected code in repl", but likely it can be done using existing RsConsoleRunnerImpl. So yes, removed